modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/ScalaLagomServerCodegen.java

```diff
@@ -0,0 +1,280 @@
+    additionalProperties.put("authScheme", authScheme);
+    additionalProperties.put("authPreemptive", authPreemptive);
+
+    supportingFiles.add(new SupportingFile("README.mustache", "", "README.md"));
+    supportingFiles.add(new SupportingFile("gitignore.mustache", "", ".gitignore"));
+    supportingFiles.add(new SupportingFile("build.sbt.mustache", "", "build.sbt"));
+    supportingFiles.add(new SupportingFile("build.properties.mustache", "", "project/build.properties"));
+    supportingFiles.add(new SupportingFile("plugins.sbt.mustache", "", "project/plugins.sbt"));
+
+    importMapping.remove("List");
+    importMapping.remove("Set");
+    importMapping.remove("Map");
+
+    importMapping.put("DateTime", "org.joda.time.DateTime");
+    importMapping.put("ListBuffer", "scala.collection.mutable.ListBuffer");
+
+    typeMapping = new HashMap<String, String>();
+    typeMapping.put("Integer", "Int");
+    typeMapping.put("enum", "NSString");
+    typeMapping.put("array", "Seq");
+    typeMapping.put("set", "Set");
+    typeMapping.put("boolean", "Boolean");
+    typeMapping.put("string", "String");
+    typeMapping.put("int", "Int");
+    typeMapping.put("long", "Long");
+    typeMapping.put("float", "Float");
+    typeMapping.put("byte", "Byte");
+    typeMapping.put("short", "Short");
+    typeMapping.put("char", "Char");
+    typeMapping.put("long", "Long");
+    typeMapping.put("double", "Double");
+    typeMapping.put("object", "Any");
+    typeMapping.put("file", "File");
+
+    //TODO binary should be mapped to byte array
+    // mapped to String as a workaround
+    typeMapping.put("binary", "String");
+    typeMapping.put("ByteArray", "String");
+
+    instantiationTypes.put("array", "ListBuffer");
+    instantiationTypes.put("map", "HashMap");
+
+    cliOptions.add(new CliOption(CodegenConstants.MODEL_PROPERTY_NAMING,
+        CodegenConstants.MODEL_PROPERTY_NAMING_DESC).defaultValue("camelCase"));
+  }
+
+  @Override
+  public void processOpts() {
+    super.processOpts();
+
+    if (additionalProperties.containsKey(CodegenConstants.MODEL_PROPERTY_NAMING)) {
+      setModelPropertyNaming(
+          (String) additionalProperties.get(CodegenConstants.MODEL_PROPERTY_NAMING));
+    }
+  }
+
+  public void setModelPropertyNaming(String naming) {
+    if ("original".equals(naming) || "camelCase".equals(naming) ||
+        "PascalCase".equals(naming) || "snake_case".equals(naming)) {
+      this.modelPropertyNaming = naming;
+    } else {
+      throw new IllegalArgumentException("Invalid model property naming '" +
+          naming + "'. Must be 'original', 'camelCase', " +
+          "'PascalCase' or 'snake_case'");
+    }
+  }
+
+  public String getModelPropertyNaming() {
+    return this.modelPropertyNaming;
+  }
+
+  @Override
+  public String toVarName(String name) {
+    // sanitize name
+    name = sanitizeName(
+        name); // FIXME: a parameter should not be assigned. Also declare the methods parameters as 'final'.
```

Comment:
This looks like a copy/paste. Can it be updated/fixed here, or does that affect something elsewhere?

```
String modifiedName = name;
```

Then, use `modifiedName` within the method?

Reply:
Jim, you are right it is a copy paste from  ScalaClientCodegen.Java. I guess this is needed for scala variables. Not sure what will break. Hence retained the logic. 
With reference to method signatures. 
Please have a look at https://github.com/lagom/sbt-lagom-descriptor-generator
The method signature generated by this PR matches with the one generated in above project. 
Please let me know if you find any gaps.